packages/react-magma-dom/src/components/Popover/PopoverContent.tsx

```diff
@@ -0,0 +1,224 @@
+      ${props =>
+        props.isInverse
+          ? props.theme.colors.primary600
+          : props.theme.colors.neutral100}
+      transparent;
+    left: 50%;
+    bottom: 100%;
+    transform: translateX(-50%) rotate(0deg);
+  }
+  &[data-popover-placement=${PopoverPosition.top}]::after {
+    border-color: ${props =>
+        props.isInverse
+          ? props.theme.colors.primary600
+          : props.theme.colors.neutral100}
+      transparent transparent transparent;
+    left: 50%;
+    top: 100%;
+    transform: translateX(-50%) rotate(0deg);
+  }
+`;
+
+const StyledAnnounce = styled(Announce)`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+`;
+
+const ErrorMessage = styled.div`
+  width: fit-content;
+  padding: 1em;
+`;
+
+const StyledPopoverContent = styled.div`
+  z-index: 2;
+`;
+
+const ScrollableContent = styled.div`
+  display: flex;
+  flex-direction: column;
+  overflow-y: auto;
+  padding: 16px;
+  height: inherit;
+  white-space: normal;
+`;
+
+export const PopoverContent = React.forwardRef<
+  HTMLDivElement,
+  PopoverContentProps
+>((props, forwardedRef) => {
+  const { children, testId, ...other } = props;
+  const context = React.useContext(PopoverContext);
+  const theme = React.useContext(ThemeContext);
+  const ref = useForkedRef(forwardedRef, context.contentRef);
+
+  const focusTrapRef = useFocusLock(
+    context.isOpen && isExistedActiveElements(context.contentRef)
+  );
+
+  const styledChildren = React.Children.toArray(children).map(item =>
+    React.cloneElement(item as React.ReactElement, {
+      theme,
+    })
+  );
+  const header = styledChildren.find(
+    child =>
+      (child as React.ReactElement).type &&
+      (child as React.ReactElement).type === PopoverHeader
+  );
+  const footer = styledChildren.filter(
+    child =>
+      !(child as React.ReactElement).type ||
+      (child as React.ReactElement).type === PopoverFooter
+  );
+  const content = styledChildren.filter(
+    child =>
+      !(child as React.ReactElement).type ||
+      ((child as React.ReactElement).type !== PopoverHeader &&
+        (child as React.ReactElement).type !== PopoverFooter)
+  );
+
+  return (
+    <StyledPopoverContent
+      ref={context.setFloating}
+      style={{ ...context.floatingStyles }}
+    >
+      <StyledCard
+        {...other}
+        position={context.position}
+        hasDropShadow
+        isInverse={context.isInverse}
+        isOpen={context.isOpen}
+        maxHeight={context.maxHeight}
+        ref={ref}
+        tabIndex={-1}
+        testId={testId || 'popoverContent'}
+        theme={theme}
+        width={context.width}
+        data-popover-placement={context.position}
+        hasPointer={context.hasPointer}
+      >
+        <div
+          aria-labelledby={context.popoverTriggerId.current}
+          role="dialog"
+          ref={focusTrapRef}
+        >
+          {!content.length ? (
+            <ErrorMessage>Content must be passed</ErrorMessage>
```

Comment:
I'm debating whether we should show an error message like this, or just show empty space. I don't think we do this for any other component, and I worry about this being visible to actual users accidentally. Just curious about your thought process in including this.